### PR TITLE
feat: adapt Transaction Delivery man to arb network

### PR DESCRIPTION
**Pull request by koonopek**

### CODE REVIEW

File: packages/on-chain-relayer/src/config.ts [hunk 1]
```diff
@@ -48,6 +48,7 @@ export const config = Object.freeze({
   expectedTxDeliveryTimeInMS: Number(
     getFromEnv("EXPECTED_TX_DELIVERY_TIME_IN_MS", false)
   ),
+  isArbitrumNetwork: getFromEnv("IS_ARBITRUM_NETWORK", true) === "true",
```
Comment (koonopek) [inline packages/on-chain-relayer/src/config.ts:51]: Maybe this name is a little bit stupid, however, my priority was to minimize errors when someone will be configuring on-chain-relayer

File: packages/rpc-providers/src/TransactionDeliveryMan.ts [hunk 2]
```diff
@@ -50,15 +60,18 @@ type TransactionDeliveryManOpts = {
...
 const unsafeBnToNumber = (bn: BigNumber) => Number(bn.toString());
 
-const getEthFeeFromGasOracle: GasOracleFn = async (apiKey: string = "") => {
+const getEthFeeFromGasOracle: GasOracleFn = async (
+  opts: TransactionDeliveryManOpts
+) => {
   const response = // rate limit is 5 seconds
     (
       await fetchWithCache<any>(
-        `https://api.etherscan.io/api?module=gastracker&action=gasoracle&apikey=${apiKey}`,
```
Comment (hatskier) [inline packages/rpc-providers/src/TransactionDeliveryMan.ts:61]: Why do we remove the apiKey?

File: packages/rpc-providers/src/TransactionDeliveryMan.ts [hunk 3]
```diff
@@ -161,8 +176,13 @@ export class TransactionDeliveryMan {
 
   private isUnderpricedError(e: any) {
     return (
+      // RPC errors sucks most of the time, thus we can really relay on them
```
Comment (hatskier) [inline packages/rpc-providers/src/TransactionDeliveryMan.ts:179]: can -> can not?

Comment (hatskier) [reply packages/rpc-providers/src/TransactionDeliveryMan.ts:179]: relay -> rely?

File: packages/rpc-providers/src/TransactionDeliveryMan.ts [hunk 4]
```diff
@@ -240,21 +272,42 @@ export class TransactionDeliveryMan {
...
     );
     const maxFeePerGas = Math.round(
-      currentFees.maxFeePerGas * this.opts.priorityFeePerGasMultiplier
-    );
-
-    this.opts.logger(
-      `Scaling fees maxFeePerGas to ${maxFeePerGas} and maxPriorityFeePerGas to ${maxPriorityFeePerGas}`
+      currentFees.maxFeePerGas * this.opts.multiplier
     );
+    const gasLimit = this.opts.twoDimensionFees
```
Comment (hatskier) [inline packages/rpc-providers/src/TransactionDeliveryMan.ts:299]: Do I understand correctly that if `twoDimensionFees` is `true`, then we increase both fee and gasLimit in the same way (using the same multiplier)?

**Review by hatskier — DISMISSED** [review]

File: packages/rpc-providers/src/TransactionDeliveryMan.ts (hunk 4 above)
Comment (koonopek) [reply packages/rpc-providers/src/TransactionDeliveryMan.ts:299]: Correct, if we observe that multiplier for gaszlimit.shoul.be Independent,we can always change it

File: packages/rpc-providers/src/TransactionDeliveryMan.ts (hunk 2 above)
Comment (koonopek) [reply packages/rpc-providers/src/TransactionDeliveryMan.ts:61]: We are using cache.so.we will.never get into api reqursts limits (we werent using it anyway)

File: packages/rpc-providers/src/TransactionDeliveryMan.ts (hunk 4 above)
Comment (koonopek) [reply packages/rpc-providers/src/TransactionDeliveryMan.ts:277]: Changed to seperate

File: packages/rpc-providers/src/TransactionDeliveryMan.ts (hunk 2 above)
Comment (hatskier) [reply packages/rpc-providers/src/TransactionDeliveryMan.ts:61]: Should we have the `&apikey=` then?

**Review by hatskier — APPROVED** [review]

File: packages/rpc-providers/src/TransactionDeliveryMan.ts (hunk 2 above)
Comment (koonopek) [reply packages/rpc-providers/src/TransactionDeliveryMan.ts:61]: Will fix in next PR :)

**Pull request merged**